src/Control/Effect/Interpret.hs

```diff
@@ -1,20 +1,74 @@
-{-# LANGUAGE GeneralizedNewtypeDeriving, MultiParamTypeClasses, RankNTypes, TypeOperators, UndecidableInstances #-}
+{-# language FlexibleContexts #-}
+{-# language FlexibleInstances #-}
+{-# language FunctionalDependencies #-}
+{-# language GeneralizedNewtypeDeriving #-}
+{-# language KindSignatures #-}
+{-# language PolyKinds #-}
+{-# language RankNTypes #-}
+{-# language ScopedTypeVariables #-}
+{-# language TypeApplications #-}
+{-# language TypeOperators #-}
+{-# language UndecidableInstances #-}
+
 module Control.Effect.Interpret
 ( runInterpret
-, InterpretC(..)
 , runInterpretState
-, InterpretStateC(..)
+, InterpretC(..)
 ) where
 
 import Control.Applicative (Alternative(..))
 import Control.Effect.Carrier
-import Control.Effect.Reader
 import Control.Effect.State
 import Control.Monad (MonadPlus(..))
 import Control.Monad.Fail
 import Control.Monad.Fix
 import Control.Monad.IO.Class
 import Control.Monad.Trans.Class
+import Unsafe.Coerce ( unsafeCoerce )
+
+
+newtype InterpretC ( s :: * ) ( sig :: ( * -> * ) -> * -> * ) m a =
```

Comment:
Could you please move `InterpretC` down after `runInterpret` and `runInterpretState`? 🙏